### PR TITLE
add ppo tests

### PR DESCRIPTION
**Pull request by budzianowski**
Added all ppo tests - caught that rewards were mistakenly using dones for masks.
Explicit env and time checking since the rewards were written originally for (time,envs).

### CODE REVIEW

File: ksim/losses/loss.py [hunk 1]
```diff
@@ -0,0 +1,13 @@
...
+from typing import Any
+
+import equinox as eqx
+from jaxtyping import Array
+
+
+class Loss(ABC):
+    @abstractmethod
+    def compute_loss(self, agent: eqx.Module, batch: Any, output: Any) -> Array:  # noqa: ANN401
+        """Compute the loss for a given model and batch."""
```
Comment (codekansas) [inline ksim/losses/loss.py]: i don't think it really makes sense to have this as our base abstract class - losses don't really follow this interface (e.g., the ppo loss below has a completely different function signature)

better to just have some sort of utility class exclusively for ppo loss, don't need to have this here

File: ksim/losses/test_ppo.py [hunk 2]
```diff
@@ -0,0 +1,115 @@
...
+                [1.9702],
+                [1.48505],
+                [0.995],
+                [0.5],
+                [1.9702],
+                [1.48505],
+                [0.995],
+                [0.5],
+            ]
+        )
```
Comment (codekansas) [inline ksim/losses/test_ppo.py]: can just write the equations in the array instead of magic numbers with comments

File: ksim/losses/ppo.py [hunk 3]
```diff
@@ -0,0 +1,175 @@
...
+            "prediction_mean": jnp.mean(prediction),
+            "prediction_std": jnp.std(prediction),
+            "log_prob_mean": jnp.mean(log_probs),
+            "log_prob_max": jnp.max(log_probs),
+            "log_prob_min": jnp.min(log_probs),
+            "values_std": jnp.std(values),
+            "values_mean": jnp.mean(values),
+            "obs_nans_ratio": xax.compute_nan_ratio(initial_action),
+            "action_nans_ratio": xax.compute_nan_ratio(initial_action),
+        }
```
Comment (codekansas) [inline ksim/losses/ppo.py]: i think we should separate this out into a different class - have this function return the relevant tensors, then have a separate function

**Comment by codekansas** [comment]
hmm i don't think this is the right way to fix nans... gonna implement a different version
